numba/listobject.py

```diff
@@ -1005,9 +1005,13 @@ def impl(l):
 
 @overload_method(types.ListType, 'copy')
 def impl_copy(l):
+    itemty = l.item_type
     if isinstance(l, types.ListType):
         def impl(l):
-            return l[:]
+            newl = new_list(itemty, len(l))
```

Comment:
As a record of an out of band conversation. This can be done as a memcpy and then if the dtype is recounted, incref all the elements.